src/modules/core/components/InvisibleCopyableAddress/InvisibleCopyableAddress.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactNode, useState, useEffect } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import copyToClipboard from 'copy-to-clipboard';
+import { Address } from '~types/index';
+import { Tooltip } from '~core/Popover';
+
+import styles from './InvisibleCopyableAddress.css';
+
+interface Props {
+  /** Children element as we don't want style address here */
+  children: ReactNode;
+  /** Address to display and copy */
+  address: Address;
+}
+
+const MSG = defineMessages({
+  copyAddressTooltip: {
+    id: 'InvisibleCopyableAddress.copyAddressTooltip',
+    defaultMessage: `{copied, select,
+      true {Copied}
+      false {Click to copy colony address}
+    }`,
+  },
+});
+
+const displayName = 'InvisibleCopyableAddress';
+
+const InvisibleCopyableAddress = ({ children, address }: Props) => {
+  const [copied, setCopied] = useState(false);
+  const handleClipboardCopy = () => {
+    setCopied(true);
+    copyToClipboard(address);
+  };
+
+  useEffect(() => {
+    let timeout;
+    if (copied) {
+      timeout = setTimeout(() => setCopied(false), 2000);
```

Comment:
maybe the timer value should be configurable via props?

Reply:
i mean, i expect consistency across the app for time for this component. So if this will be too long/too short i would just adjust here. but apply for all of them 🤔